### PR TITLE
chore(post-processor): update `vsphere-template`

### PR DESCRIPTION
**Pull request by tenthirtyam**
### Summary

- Updates error messages to use a more well-formed structure.
- Updates and simplifies code comments.
- Updates post-processor to generate documentation from structs.
- Generates the updated post-processor documentation.

### Testing

```console
packer-plugin-vsphere on  chore(post-processor)/update-vsphere-template [$!?] via 🐹 v1.22.1 
➜ make dev


packer-plugin-vsphere on  chore(post-processor)/update-vsphere-template [$!?] via 🐹 v1.22.1 
➜ make generate
2024/03/14 14:01:06 Copying "docs" to ".docs/"
2024/03/14 14:01:06 Replacing @include '...' calls in .docs/
Compiling MDX docs in '.docs' to Markdown in '.web-docs'...


packer-plugin-vsphere on  chore(post-processor)/update-vsphere-template [$!?] via 🐹 v1.22.1 took 11.6s 
➜ make build


packer-plugin-vsphere on  chore(post-processor)/update-vsphere-template [$!?] via 🐹 v1.22.1 took 3.2s 
➜ make test
?       github.com/hashicorp/packer-plugin-vsphere      [no test files]
?       github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common/testing       [no test files]
?       github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/examples/driver      [no test files]
?       github.com/hashicorp/packer-plugin-vsphere/version      [no test files]
ok      github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/clone        2.092s
ok      github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common       2.645s
ok      github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/driver       6.430s
ok      github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/iso  1.438s
ok      github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/supervisor   4.679s
ok      github.com/hashicorp/packer-plugin-vsphere/post-processor/vsphere       1.781s
ok      github.com/hashicorp/packer-plugin-vsphere/post-processor/vsphere-template      2.089s
```

### CODE REVIEW

**Review by lbajolet-hashicorp — APPROVED** [review]
LGTM overall @tenthirtyam, I just left a couple nits, but the changes are good to go aside those

File: post-processor/vsphere-template/post-processor.go [hunk 1]
```diff
@@ -40,19 +38,37 @@ var builtins = map[string]string{
 	"packer.post-processor.artifice": "artifice",
 }
 
+type RequiredConfig struct {
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply post-processor/vsphere-template/post-processor.go:41]: doh! lemme fix that! 🤦🏻‍♂️

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply post-processor/vsphere-template/post-processor.go:41]: that will be so much simpler

[the remark this answers is not in the document]
Comment (tenthirtyam) [reply post-processor/vsphere-template/post-processor.go]: Resolved in https://github.com/hashicorp/packer-plugin-vsphere/commit/86108893ee1547db4c6280df59be2571639e146e.

File: .web-docs/components/post-processor/vsphere-template/README.md [hunk 2]
```diff
@@ -1,64 +1,61 @@
...
-template, and places it in the path of your choice.
+This post-processor uses an artifact from the `vmware-iso` builder with an ESXi host or an artifact
+from the [vSphere](/packer/integrations/hashicorp/vsphere/latest/components/post-processor/vsphere) post-processor. It then marks
+the virtual machine as a template and moves it to your specified path.
 
 ## Configuration
 
-There are many configuration options available for the post-processor. They are
-provided below in two categories: required and optional parameters.
+The following configuration options available for the post-processor.
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply .web-docs/components/post-processor/vsphere-template/README.md]: Resolved in https://github.com/hashicorp/packer-plugin-vsphere/commit/86108893ee1547db4c6280df59be2571639e146e.

**Review by lbajolet-hashicorp — APPROVED** [review]
LGTM!

**Pull request merged**